common/common-client/src/main/java/org/ow2/proactive/utils/console/MBeanInfoViewer.java

```diff
@@ -138,10 +224,59 @@ public void invoke(final String mbeanNameAsString, final String operationName, f
         }
     }
 
+    public String retrieveStats(String mbeanName, String range, String[] dataSources) {
+        lazyConnect();
+
+        try {
+            AttributeList attrs = connection.getAttributes(new ObjectName(mbeanName),
+                                                           new String[] { "StatisticHistory" });
+            Attribute attr = (Attribute) attrs.get(0);
+            // content of the RRD4J database backing file
+            byte[] rrd4j = (byte[]) attr.getValue();
+
+            return MBeanInfoViewer.rrdContent(rrd4j,
+                                              MBeanInfoViewer.possibleModifyRange(range, dataSources, 'd'),
+                                              dataSources);
+        } catch (Exception e) {
+            LOGGER.error("Could not retrieve statistics history, " + e.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static long secondsInZone(char zone) {
+        switch (zone) {
+            default:
+            case 'a': // 1 minute
+                return 60;
+            case 'n': // 5 minutes
+                return 60 * 5;
+            case 'm': // 10 minutes
+                return 60 * 10;
+            case 't': // 30 minutes
+                return 60 * 30;
+            case 'h': // 1 hour
+                return 60 * 60;
+            case 'j': // 2 hours
+                return 60 * 60 * 2;
+            case 'k': // 4 hours
+                return 60 * 60 * 4;
+            case 'H': // 8 hours
+                return 60 * 60 * 8;
+            case 'd': // 1 day
+                return 60 * 60 * 24;
+            case 'w': // 1 week
+                return 60 * 60 * 24 * 7;
+            case 'M': // 1 month
+                return 60 * 60 * 24 * 28;
```

Comment:
1 month => 4 weeks

Reply:
oook. it was there before. I would keep it..